geoarrow-types/src/geoarrow/types/type_pyarrow.py

```diff
@@ -0,0 +1,314 @@
+        return pa.field(names[-1], inner_type, nullable=False)
+
+
+def _nested_type(coord, names):
+    if len(names) > 0:
+        return pa.list_(_nested_field(coord, names))
+    else:
+        return coord
+
+
+def _append_fingerprint(obj, fingerprint):
+    if isinstance(obj, pa.Field):
+        fingerprint.append(obj.name)
+        _append_fingerprint(obj.type, fingerprint)
+        return
+
+    fingerprint.append(obj.id)
+    if pa_types.is_list(obj):
+        _append_fingerprint(obj.value_type, fingerprint)
+    elif pa_types.is_struct(obj):
+        for i in range(obj.num_fields):
+            _append_fingerprint(obj.field(i), fingerprint)
+
+    return fingerprint
+
+
+def _generate_storage_types():
+    coord_storage = {
+        (CoordType.SEPARATE, Dimensions.XY): _struct_fields("xy"),
+        (CoordType.SEPARATE, Dimensions.XYZ): _struct_fields("xyz"),
+        (CoordType.SEPARATE, Dimensions.XYM): _struct_fields("xym"),
+        (CoordType.SEPARATE, Dimensions.XYZM): _struct_fields("xyzm"),
+        (CoordType.INTERLEAVED, Dimensions.XY): _interleaved_fields("xy"),
+        (CoordType.INTERLEAVED, Dimensions.XYZ): _interleaved_fields("xyz"),
+        (CoordType.INTERLEAVED, Dimensions.XYM): _interleaved_fields("xym"),
+        (CoordType.INTERLEAVED, Dimensions.XYZM): _interleaved_fields("xyzm"),
+    }
+
+    field_names = {
+        GeometryType.POINT: [],
+        GeometryType.LINESTRING: ["vertices"],
+        GeometryType.POLYGON: ["rings", "vertices"],
+        GeometryType.MULTIPOINT: ["points"],
+        GeometryType.MULTILINESTRING: ["linestrings", "vertices"],
+        GeometryType.MULTIPOLYGON: ["polygons", "rings", "vertices"],
+    }
+
+    all_geoemetry_types = list(field_names.keys())
+    all_coord_types = [CoordType.INTERLEAVED, CoordType.SEPARATE]
+    all_dimensions = [Dimensions.XY, Dimensions.XYZ, Dimensions.XYM, Dimensions.XYZM]
+
+    all_storage_types = {}
+    for geometry_type in all_geoemetry_types:
+        for coord_type in all_coord_types:
+            for dimensions in all_dimensions:
+                names = field_names[geometry_type]
+                coord = coord_storage[(coord_type, dimensions)]
+                key = geometry_type, dimensions, coord_type
+                storage_type = _nested_type(coord, names)
+                all_storage_types[key] = storage_type
+
+    return all_storage_types
+
+
+_EXTENSION_CLASSES = {
+    "geoarrow.wkb": WkbType,
+    "geoarrow.wkt": WktType,
+    "geoarrow.point": PointType,
+    "geoarrow.linestring": LinestringType,
+    "geoarrow.polygon": PolygonType,
+    "geoarrow.multipoint": MultiPointType,
+    "geoarrow.multilinestring": MultiLinestringType,
+    "geoarrow.multipolygon": MultiPolygonType,
+}
+
+_SERIALIZED_STORAGE_TYPES = {
+    Encoding.WKT: pa.utf8(),
+    Encoding.LARGE_WKT: pa.large_utf8(),
+    Encoding.WKB: pa.binary(),
+    Encoding.LARGE_WKB: pa.large_binary(),
+}
+
+_NATIVE_STORAGE_TYPES = _generate_storage_types()
```

Comment:
I would personally expose those storage type definitions a bit more publicly. Although I assume one can of course do something like `PointType().storage_type` ?

For example in geopandas, the main thing I currently need are those storage type definitions (I don't know if it would be worth it for geopandas to depend on this package if it is just for that, though, but it's probably a good example use case)